app/controllers/carts_controller.rb

```diff
@@ -1,19 +1,24 @@
 # frozen_string_literal: true
 
 class CartsController < ApplicationController
-  def show
-    @cart = Cart.find_by(user_id: current_user.id, status: 'Pending')
-  end
-
-  def add
-    @cart = Cart.find_by(user_id: current_user.id, status: 'Pending')
+  before_action :create_new_cart
+  
+  
+  def create_new_cart
     if @cart.nil?
```

Comment:
The variable `@cart` has not been defined before this line is run, so this will always return `true`. 

A better way of dealing with this issue would be to check to see if the current user has a cart by adding a `has_cart?` method to the user controller and then calling `current_user.has_cart?` here.

Reply:
So there's not a users controller currently as devise does that in the background. I could generate it, but that would bring in a lot more that I would have to work on to get that right. I put the logic that I used in the each of the actions that needed it as I had it before.

Reply:
Marcus, this code will not do what you want.  It will create a new cart for every action to this controller.  You need to change it to do a find for the cart, and then if the cart is still nil (because it was not found), it should create the cart.  Then you could take the call to find the cart out of many of the subsequent methods.